validation-test/IDE/crashers_2_fixed/sr14494.swift

```diff
@@ -0,0 +1,37 @@
+// RUN: %target-swift-ide-test -code-completion -code-completion-token=COMPLETE -source-filename=%s | %FileCheck %s
+
+protocol MyView {
+    associatedtype Body : MyView
+    @MyViewBuilder var body: Self.Body { get }
+}
+
+@resultBuilder public struct MyViewBuilder {
+    static func buildBlock() -> MyZStack<Never> { fatalError() }
+    static func buildBlock<Content>(_ content: Content) -> Content { content }
+}
+
+struct MyAlignment {
+  static let center: MyAlignment
+}
+
+struct MyZStack<Content> : MyView {
+    init(alignment: MyAlignment, @MyViewBuilder content: () -> Content) {
+        fatalError()
+    }
+
+    func my_updating<State>(body: (inout State) -> Void) {}
+}
+
+struct BottomMenu: MyView {
+    var body: some MyView {
+        let a = MyZStack(alignment: .#^COMPLETE^#center, content: {})
+        .my_updating(body: { state in
```

Comment:
So here is the interesting thing about this example - original type-check of this expression succeeds and produces a well-formed call. But then `typeCheckForCodeCompletion` called on type-checked AST and it seems not all of the information is erased. Why does that expression get re-typechecked that's probably where the investigation should start...